stories/10-For Developers/Token Implementation Notes.stories.mdx

```diff
@@ -0,0 +1,229 @@
+import { Meta } from '@storybook/addon-docs/blocks';
+
+<Meta title="For Developers/Token Implementation Notes" />
+
+Tokens are placeholders for specific values in our CSS. For example, instead of specifiying text color as a dark gray, we can simply say it's the "text color" and let each theme fill in the details.
+
+Tokens are what power Radial's themability. By building our components with tokens instead of specific values, we allow them to be easily rethemed. Tokens won't cover everything we want in a new design but they let us tune the defaults in a way that resembles configuration more than coding.
+
+## Glossary
+
+There are a few terms around tokens that can be confusing so let's define them up front.
+
+**Variables:** A name and a value. Here we're usually talking about CSS custom properties, also known as CSS variables. This is the technical feature we use to implement tokens.
+
+**Tokens:** CSS Variables that a developer uses when implementing the base styles for a component. Not all variables are tokens. For example, a theme might have a local palette of colors that's used to set the token values. When I write some code for a theme that defines "button color" as "brand red", "button color" is a token, but "brand red" isn't.
+
+**Dials:** Variables that can be set by a theme author to change the value of a single token. For example, "button color".
+
+**Levers:** Variables that can be set by a theme author to change the value of multiple tokens. For example, the font scaling factor which changes all of the derived font size tokens.
+
+## CSS Custom Properties
+
+We chose to go with CSS custom properties over SASS variables because of several features that weren't easily replicated with SASS.
+
+Advantages
+- Preserving the variable names in source allows us to use the CSS as a source of truth in our storybook docs.
+- Being able to see the variable names in the browser devtools makes debugging more transparent.
+- CSS's scoping rules makes implementing color groups using CSS variables pretty straightforward.
+
+Disadvantages:
+- They have limited support in IE.
+
+## Token Linting
+
+We use stylelint to enforce an allowlist so that things like theme specific colors don't get added to component styles.
+
+Right now we only apply this check to the base styles and are less strict with what goes into the site specific themes themselves. There's no benifit to saying "brand theme" can't use "brand color" directly.
+
+## Component Specific Tokens
+
+TODO: Add recommendations for adding dials to components. This will probably be detailed more in it's own doc or a doc about components.
```

Comment:
```suggestion
TODO: Add recommendations for adding dials to components. This will probably be detailed more in its own doc or a doc about components.
```